### PR TITLE
Add link to ROS QoS Policy docs to spec registry

### PR DESCRIPTION
**Pull request by defunctzombie**
Update the `ros2` profile section of the registry with more details about the expectations for `offered_qos_profiles` and link to the official ROS docs.

Fixes: #1002

---
I do wonder if it would be more appropriate for the ROS docs to define these aspects of the profile and the MCAP registry links to those docs for details on the profile.

### CODE REVIEW

**Comment by defunctzombie** [comment]
@MichaelOrlov @jhurliman fyi - if you have feedback.

**Review by achim-k — APPROVED** [review]

**Comment by MichaelOrlov** [comment]
@defunctzombie My apologies, a bit busy and don't have time to consider this PR.  
However, I can get back to it after Thanksgiving holidays.

**Comment by jhurliman** [comment]
LGTM

**Review by MichaelOrlov — COMMENTED** [review]
@defunctzombie The wording and links looks good to me.

However, there are some subtle specifics that in theory could be more than one QoS profile stored in that field and it is actually a sequence of such serialized QoS profiles in yaml format. This is true for the case when exists different publishers with different QoS profiles on the same topic.
See for details https://github.com/ros2/rosbag2/blob/9855072566f8fb46a2bdd20cc217d1a0c2b03e82/rosbag2_storage_mcap/src/mcap_storage.cpp#L823-L825
```
    channel.metadata.emplace(
      "offered_qos_profiles",
      rosbag2_storage::serialize_rclcpp_qos_vector(topic_info.topic_metadata.offered_qos_profiles));
```
Maybe reword the first sentence `The sequence of the QoS policy values for the topic in yaml format`?

**Comment by defunctzombie** [comment]
> Maybe reword the first sentence `The sequence of the QoS policy values for the topic in yaml format`?

👍 sounds good. Do you know what the format of this sequence would look like? Would it be a typical yaml sequence of the existing QoS "objects"? The above code makes it look like it will always be a sequence even with 1 element. Is that correct or does the serialization of the vector special case single element vectors?

**Comment by MichaelOrlov** [comment]
It is going to be always a YAML sequence.
The implementation is https://github.com/ros2/rosbag2/blob/542bda34358d03acf3c15228a60859e62f5b69ba/rosbag2_storage/src/rosbag2_storage/qos.cpp#L252-L259
```
Node convert<std::vector<rclcpp::QoS>>::encode(const std::vector<rclcpp::QoS> & rhs, int version)
{
  Node node{NodeType::Sequence};
  for (const auto & value : rhs) {
    node.push_back(convert<rclcpp::QoS>::encode(value, version));
  }
  return node;
}
```
